lib/runner/TestRunner.js

```diff
@@ -57,62 +57,77 @@ module.exports = class TestRunner {
             }
 
             // Process through the interactions
-            // Look for goto statements and exits, which will affect the sequence
-            let goto;
-            for (const interaction of test.interactions) {
-                // If a goto is set, keep skipping until we match it
-                if (goto) {
-                    if (goto === interaction.utterance) {
-                        goto = undefined;
-                    } else {
-                        continue;
-                    }
-                }
-
-                const interactionResult = await this.interact(invoker, testSuite, test, interaction);
-                if (interactionResult.goto) {
-                    // If this result is a goto, set the goto label
-                    goto = interactionResult.goto;
-                }
-
-                testResult.addInteractionResult(interactionResult);
+            // If there is a goto or this is not a batch invoker, run them one-by-one
+            let results;
+            if (test.hasGoto || !invoker.batchSupported()) {
+                results = await this.sequentialRun(invoker, testSuite, test.interactions);
 
-                if (interactionResult.exited) {
-                    // If this is an exit, stop processing
-                    break;
-                }
+            } else {
+                results = await this.batchRun(invoker, testSuite, test.interactions);
             }
 
+            testResult.interactionResults = results;
             invoker.afterTest(test);
         }
 
         invoker.after(testSuite);
         return testResults;
     }
 
-    async interact(invoker, testSuite, test, interaction) {
-        if (testSuite.trace) {
-            // eslint-disable-next-line no-console
-            console.log("File: " +  testSuite.shortFileName + " Test: " + test.description + " Utterance: " + interaction.utterance);
+    async batchRun(invoker, testSuite, interactions) {
+        const responses = await this.invokeBatch(interactions);
+        responses.forEach((response) => response.inject());
+
+        const interactionResults = [];
```

Comment:
since you're already using forEach we can use map to simplify the code

return responses.map((response) => this.processResponse(response));

this covers from 81 to 85